### PR TITLE
Add basic authentication management

### PR DESCRIPTION
**Pull request by TomChv**
# Description

There are many changes so here's an agnostic list
- Clean up architecture
- Setup session
- Add logger
- Add clean error
- Add authentication guard
- Clean import path
- Update lister
- Add role

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

It completely refactor the API and a new Postman workflow is required
/cc @Cleopha You want to take care of it?

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [ ] I have tested this code
- [x] I have added sufficient documentation in the code

### CODE REVIEW

File: .github/workflows/backend-ci.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: Continuous Integration Backend
+
+on:
+  push:
+    branches: ["main", "master", "develop"]
+  pull_request:
+    branches: ["main", "master", "develop"]
```
Comment (0xpanoramix) [inline .github/workflows/backend-ci.yml]: We don't have a master branch.
If you consider using master, please adjust accordindly.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,22 +7,22 @@ on:
     branches: ["main", "master", "develop"]
```
Comment (0xpanoramix) [inline .github/workflows/build.yml]: Same as in the backend ci, we don't have a master branch

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -7,22 +7,22 @@ on:
...
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        build_type: [Debug, Release]
 
-        name: Build Area
-        runs-on: ${{ matrix.os }}
+    name: Build Area
+    runs-on: ${{ matrix.os }}
```
Comment (0xpanoramix) [inline .github/workflows/build.yml]: Please remove the matrix for the OS, as you only use Ubuntu

File: .gitignore [hunk 4]
```diff
@@ -1,12 +1,14 @@
...
 
+# Dependencies
 node_modules/
 /.pnp
 .pnp.js
 
-*.js
-
+# Build
 dist/
```
Comment (0xpanoramix) [inline .gitignore:14]: Please consider using [the full NextJS gitignore file](https://github.com/akiran/nextjs-demo/blob/master/.gitignore)

File: backend/src/session/session.module.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+			session: {
+				store,
+				secret: config.get<string>('api.session.secret' as keyof IConfig),
+				resave: true,
+				saveUninitialized: true,
+				cookie: {
+					secure: false,
+					sameSite: false,
+					httpOnly: true,
+					maxAge: 60000 * 60 * 6,
```
Comment (0xpanoramix) [inline backend/src/session/session.module.ts]: Please consider using an ENV variable for this, with the `6000 * 60 * 6` value as the default one.

**Review by 0xpanoramix — CHANGES_REQUESTED** [review]
Great job :100:
Take a look at the requested changes and feel free to merge !

Note : Next time, please submit multiple PRs for each backend major features ;)

**Review by Cleopha — APPROVED** [review]
Nice code but should reduce the next pr

File: backend/src/session/session.module.ts (hunk 5 above)
Comment (TomChv) [reply backend/src/session/session.module.ts]: Fixed in https://github.com/Cleopha/ifttt-like/pull/1/commits/bcbba3317a65cfb2873f47a75ea3928771a9d94c

**Review by 0xpanoramix — APPROVED** [review]
LGTM !

**Pull request merged**